pages/people/[id]/case-status/add/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { useRouter } from 'next/router';
+import { Form, Formik, FormikHelpers, FormikValues } from 'formik';
+import { generateInitialValues } from 'lib/utils';
+import { generateFlexibleSchema } from 'lib/validators';
+import CASE_STATUS from 'data/flexibleForms/caseStatus';
+import FlexibleField from 'components/FlexibleForms/FlexibleFields';
+import PersonView from 'components/PersonView/PersonView';
+import Button from 'components/Button/Button';
+import Link from 'next/link';
+
+const AddNewCaseStatus = (): React.ReactElement => {
```

Comment:
What do you think about extracting this into a component so we can have more confidence that we've correctly configured the form? Like how we've done with the add relationship form, we can then have unit tests are conditional viewing, validation, disabled buttons etc.

Reply:
That's actually not a bad idea, I'll move everything to a component

Reply:
Content extracted into components and tests added 😄